normandy/recipes/management/commands/update_addon_urls.py

```diff
@@ -33,17 +39,23 @@ def handle(self, *args, new_hostname, **options):
                 )
                 continue
 
-            parsed_url = urlparse(arguments["addonUrl"])
-            if parsed_url.netloc == new_hostname:
+            filename = get_filename_from_url(arguments["addonUrl"])
+
+            if filename not in extension_by_filename:
+                self.stderr.write(
+                    f"Warning: Recipe {rev.recipe.id} revision {rev.id} has an addonUrl that does "
+                    f"not match any in the database.",
+                    ending="\n",
+                )
+                continue
+
+            extension = extension_by_filename[filename]
+            new_url = extension.xpi.url
+            if arguments["addonUrl"] == new_url:
                 # nothing to do
                 continue
 
-            # _replace is not a private method. parsed_url is a named tuple,
-            # and named tuples expose all their public methods with underscores
-            # to avoid colliding with data keys.
-            new_url_parts = parsed_url._replace(netloc=new_hostname)
-            new_url = urlunparse(new_url_parts)
-            arguments["addonUrl"] = new_url
+            arguments["addonUrl"] = extension.xpi.url
```

Comment:
At this point that precious string `"adonUrl"` has been hardcoded 4 times. Probably best to turn it into a constant before the loop starts.

Reply:
I'm very tempted to add pseudo-models to arguments at some point. I wonder how bad that would be...

Amusingly, you typed `"adonUrl"`, which is a typo of the string in the code. That sort of proves your point though. I'll change it.

Reply:
I don't know what a "pseudo-model" is.

Reply:
So in Django, we have models right? Which are bundles of logic and behavior associated with some data from the database.

I'd like to make something similar that we can give the data from the arguments json blob that has methods and property accessors and is generally smarter. But it would **not** be backed by the database. So it is like a model, but not quite. So here pseudo- is used to imply it isn't a true model, but just mimicking one.